src/ai_hawk/linkedIn_easy_applier.py

```diff
@@ -844,10 +857,9 @@ def _save_questions_to_json(self, question_data: dict) -> None:
                     logger.error("JSON decoding failed")
                     data = []
 
-                # Use the new function to check for existing questions
-                question_exists = self._question_exists_in_data(question_data['question'], data)
+                should_be_saved: bool = not question_already_exists_in_data(question_data['question'], data) or self.answer_contians_company_name(question_data['answer'])
```

Comment:
shouldn't this be AND instead of OR? And shouldn't it be NOT self.answer_contians_company_name?

Reply:
Yes fixed